src/purifier.js

```diff
@@ -612,35 +619,43 @@ export function rewriteAttributesForElement(
 /**
  * If (tagName, attrName) is a CDN-rewritable URL attribute, returns the
  * rewritten URL value. Otherwise, returns the unchanged `attrValue`.
- * @see resolveUrlAttr for rewriting rules.
- * @param {string} tagName
- * @param {string} attrName
+ * See resolveUrlAttr() for rewriting rules.
+ * @param {string} tagName Lowercase tag name.
+ * @param {string} attrName Lowercase attribute name.
  * @param {string} attrValue
  * @return {string}
- * @private Visible for testing.
+ * @private
+ * @visibleForTesting
  */
 export function rewriteAttributeValue(tagName, attrName, attrValue) {
-  const tag = tagName.toLowerCase();
-  const attr = attrName.toLowerCase();
-  if (attr == 'src' || attr == 'href' || attr == 'srcset') {
-    return resolveUrlAttr(tag, attr, attrValue, self.location);
+  if (isUrlAttribute(attrName)) {
+    return resolveUrlAttr(tagName, attrName, attrValue, self.location);
   }
   return attrValue;
 }
 
+/**
+ * @param {string} attrName Lowercase attribute name.
```

Comment:
We could, but it would result in extra `toLowerCase()` calls compared to doing it once at a higher level. Also, the regex would need to be `/^(src|href|srcset)$/i` and is more prone to error.